samza-core/src/main/scala/org/apache/samza/container/SamzaContainer.scala

```diff
@@ -469,18 +469,25 @@ object SamzaContainer extends Logging {
       info("Got store consumers: %s" format storeConsumers)
 
       var loggedStorageBaseDir: File = null
+      val jobNameAndId = (
+        config.getName.getOrElse(throw new ConfigException("Missing required config: job.name")),
+        config.getJobId.getOrElse("1")
+      )
+
       if(System.getenv(ShellCommandConfig.ENV_LOGGED_STORE_BASE_DIR) != null) {
-        val jobNameAndId = (
-          config.getName.getOrElse(throw new ConfigException("Missing required config: job.name")),
-          config.getJobId.getOrElse("1")
-        )
         loggedStorageBaseDir = new File(System.getenv(ShellCommandConfig.ENV_LOGGED_STORE_BASE_DIR)
           + File.separator + jobNameAndId._1 + "-" + jobNameAndId._2)
       } else {
-        warn("No override was provided for logged store base directory. This disables local state re-use on " +
-          "application restart. If you want to enable this feature, set LOGGED_STORE_BASE_DIR as an environment " +
-          "variable in all machines running the Samza container")
-        loggedStorageBaseDir = defaultStoreBaseDir
+        config.getLoggedStoreBaseDir match {
```

Comment:
let's do a "else if .. else" instead of else { .. match case..}. The flow is simpler to understand. Thanks.